entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -543,9 +555,78 @@ public void total(
     responseObserver.onCompleted();
   }
 
+  @Override
+  public void deleteEntities(
+      DeleteEntitiesRequest request, StreamObserver<DeleteEntitiesResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    if (validateDeleteEntitiesRequest(request, requestContext)) {
+      responseObserver.onError(new ServiceException("Invalid request provided"));
+      return;
+    }
+
+    List<String> entityIds = getEntityIdsToDelete(requestContext, request);
+    if (entityIds.size() == 0) {
+      LOG.info("{}. No entities found to delete", request);
```

Comment:
if we're treating this as a success response, should make this a debug log. Debatable whether it should error out since this is likely an unintended request.

Reply:
Could be that services might be sending just the filters without knowing if there are valid entities or not. Empty response is still valid response in those cases

Reply:
That's what I was calling debatable. If the caller doesn't know it was valid or not, then an error told him it's invalid - the same as it would if it tried to delete by ID and that ID did not exist. It could be debated either way, so it's more important to be consistent with other APIs.